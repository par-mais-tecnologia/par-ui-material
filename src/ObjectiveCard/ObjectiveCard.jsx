import React, { PureComponent } from 'react'
import { Grid, Paper, SeeTheme, Typography } from '../index'
import { getCurrencyFormat } from '../Core/masks'
import * as PropTypes from 'prop-types'

export default class ObjectiveCard extends PureComponent {
  getObjectiveType (type) {
    switch (type) {
      case 'EmergencyReserve':
        return {
          title: 'RESERVA DE EMERGÊNCIA',
          titleProps: { color: SeeTheme.styles.colors.siren, marginLeft: '5px', marginTop: '2px' },
          subTitle: 'Valor total',
          alt: 'Reserva de Emergência',
          iconSrc: 'https://s3-sa-east-1.amazonaws.com/static.parmais.com.br/images/padlock.svg',
          width: 21,
          heigth: 21
        }
      case 'DreamsAndGoals':
        return {
          title: 'SONHOS E METAS',
          titleProps: { color: SeeTheme.styles.colors.siren, marginLeft: '5px', marginTop: '2px' },
          subTitle: 'Valor total',
          alt: 'Sonhos e Metas',
          iconSrc: 'https://s3-sa-east-1.amazonaws.com/static.parmais.com.br/images/clock.svg',
          width: 27,
          heigth: 27
        }
      case 'LivingInCome':
        return {
          title: 'VIVER DE RENDA',
          titleProps: { color: SeeTheme.styles.colors.siren, marginLeft: '5px' },
          subTitle: 'Valor total',
          alt: 'Viver de Renda',
          iconSrc: 'https://s3-sa-east-1.amazonaws.com/static.parmais.com.br/images/pig.svg',
          width: 30,
          heigth: 30
        }
      case 'UnderMeasure':
        return {
          title: 'INVESTIR MELHOR',
          titleProps: { color: SeeTheme.styles.colors.siren, marginLeft: '5px', marginTop: '1px' },
          subTitle: 'Valor total',
          alt: 'Reserva de Emergência',
          iconSrc: 'https://s3-sa-east-1.amazonaws.com/static.parmais.com.br/images/wallet.svg',
          width: 23,
          height: 23
        }
      default:
        return null
    }
  }

  render () {
    const { title, subTitle, value, alt, iconSrc, width, height, titleProps, objectiveType, classes } = this.props

    let standardType = null
    if (objectiveType) {
      standardType = this.getObjectiveType(objectiveType)
    }

    let titleValue = standardType ? standardType.title : title
    let subTitleValue = standardType ? standardType.subTitle : subTitle
    let titlePropsValue = standardType ? standardType.titleProps : titleProps
    let altValue = standardType ? standardType.alt : alt
    let iconSrcValue = standardType ? standardType.iconSrc : iconSrc
    let widthValue = standardType ? standardType.width : width
    let heightValue = standardType ? standardType.height : height

    return (
      <Paper elevation={1} className={`ph4 pv4 flex-column ${classes}`}>
        <Grid container
          direction='row'>
          <Grid item>
            <img
              border='0'
              alt={altValue}
              src={iconSrcValue}
              width={widthValue}
              height={heightValue} />
          </Grid>
          <Grid item>
            <Typography variant={'headline'}
              style={{ ...titlePropsValue }}>
              { titleValue }
            </Typography>
          </Grid>
        </Grid>
        <div>
          <Typography variant={'body2'}>
            { subTitleValue }
          </Typography>
        </div>
        <div>
          <Typography variant={'body1'}>
            R$ {getCurrencyFormat(value)}
          </Typography>
        </div>
      </Paper>
    )
  }
}

ObjectiveCard.propTypes = {
  title: PropTypes.string,
  subTitle: PropTypes.string,
  value: PropTypes.number,
  alt: PropTypes.string,
  iconSrc: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  titleProps: PropTypes.object,
  objectiveType: PropTypes.string,
  classes: PropTypes.string
}
