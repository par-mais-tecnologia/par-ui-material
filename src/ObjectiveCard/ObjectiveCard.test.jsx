import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ObjectiveCard from './'

configure({ adapter: new Adapter() })

export const test = describe('ObjectiveCard', () => {
  it('shallow ObjectiveCard', () => {
    const wrapper = shallow(
      <ObjectiveCard />
    )
    expect(wrapper).toBeTruthy()
  })

  it('shallow ObjectiveCard match Snapshot', () => {
    const wrapper = shallow(
      <ObjectiveCard />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
