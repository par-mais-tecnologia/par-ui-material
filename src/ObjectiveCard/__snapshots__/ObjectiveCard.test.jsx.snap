// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectiveCard shallow ObjectiveCard match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ObjectiveCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Grid
          container={true}
          direction="row"
        >
          <Grid
            item={true}
          >
            <img
              border="0"
            />
          </Grid>
          <Grid
            item={true}
          >
            <Typography
              style={Object {}}
              variant="h2"
            />
          </Grid>
        </Grid>,
        <div>
          <Typography
            style={
              Object {
                "fontSize": "16px",
              }
            }
            variant="caption"
          />
        </div>,
        <div>
          <Typography
            variant="body2"
          >
            0
          </Typography>
        </div>,
      ],
      "className": "ph4 pv4 flex-column undefined",
      "elevation": 1,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Grid
              item={true}
            >
              <img
                border="0"
              />
            </Grid>,
            <Grid
              item={true}
            >
              <Typography
                style={Object {}}
                variant="h2"
              />
            </Grid>,
          ],
          "container": true,
          "direction": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                border="0"
              />,
              "item": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": undefined,
                "border": "0",
                "height": undefined,
                "src": undefined,
                "width": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Typography
                style={Object {}}
                variant="h2"
              />,
              "item": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "style": Object {},
                "variant": "h2",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Typography
            style={
              Object {
                "fontSize": "16px",
              }
            }
            variant="caption"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "fontSize": "16px",
            },
            "variant": "caption",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Typography
            variant="body2"
          >
            0
          </Typography>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": 0,
            "variant": "body2",
          },
          "ref": null,
          "rendered": 0,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Grid
            container={true}
            direction="row"
          >
            <Grid
              item={true}
            >
              <img
                border="0"
              />
            </Grid>
            <Grid
              item={true}
            >
              <Typography
                style={Object {}}
                variant="h2"
              />
            </Grid>
          </Grid>,
          <div>
            <Typography
              style={
                Object {
                  "fontSize": "16px",
                }
              }
              variant="caption"
            />
          </div>,
          <div>
            <Typography
              variant="body2"
            >
              0
            </Typography>
          </div>,
        ],
        "className": "ph4 pv4 flex-column undefined",
        "elevation": 1,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Grid
                item={true}
              >
                <img
                  border="0"
                />
              </Grid>,
              <Grid
                item={true}
              >
                <Typography
                  style={Object {}}
                  variant="h2"
                />
              </Grid>,
            ],
            "container": true,
            "direction": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  border="0"
                />,
                "item": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": undefined,
                  "border": "0",
                  "height": undefined,
                  "src": undefined,
                  "width": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Typography
                  style={Object {}}
                  variant="h2"
                />,
                "item": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                  "style": Object {},
                  "variant": "h2",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Typography
              style={
                Object {
                  "fontSize": "16px",
                }
              }
              variant="caption"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "style": Object {
                "fontSize": "16px",
              },
              "variant": "caption",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Typography
              variant="body2"
            >
              0
            </Typography>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 0,
              "variant": "body2",
            },
            "ref": null,
            "rendered": 0,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
